infrastructure/modules/aws/eks/iam.tf

```diff
@@ -1,5 +1,9 @@
 # SSM managed instance core
-resource "aws_iam_role_policy_attachment" "ssm_agent" {
-  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
-  role       = module.eks.worker_iam_role_name
-}
\ No newline at end of file
+# resource "aws_iam_role_policy_attachment" "ssm_agent" {
+#   for_each = module.eks.self_managed_node_groups
+#   policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+#   role       = each.value.iam_role_name
+# }
```

Comment:
I'm not sure we want to keep it commented. I would suggest to remove it entirely

Reply:
yes sure it was only intended for helping the review

Reply:
removed